### PR TITLE
impl tracker and remove song session post

### PR DESCRIPTION
**Pull request by xavluiz**
this introduces the swdc tracker for editor and codetime events (only if codetime is not installed)

it also removes the song session post call

### CODE REVIEW

**Review by bjacobson26 — COMMENTED** [review]
should add /dist to the .gitignore

File: src/KpmRepoManager.ts [hunk 1]
```diff
@@ -1,185 +1,179 @@
...
-        .replace(/\r\n/g, "\r")
-        .replace(/\n/g, "\r")
-        .replace(/^\s+/g, " ")
-        .replace(/</g, "")
-        .replace(/>/g, "")
-        .split(/\r/);
-    return resultList;
+  let result = await wrapExecPromise(cmd, projectDir);
+  if (!result) {
+    // something went wrong, but don't try to parse a null or undefined str
```
Comment (bjacobson26) [inline src/KpmRepoManager.ts:37]: should we console.log an error or ok to silently fail?

File: src/KpmRepoManager.ts [hunk 2]
```diff
@@ -1,185 +1,179 @@
...
+    return null;
+  }
+  result = result.trim();
+  let resultList = result
```
Comment (bjacobson26) [inline src/KpmRepoManager.ts:41]: can you leave a comment on what this should look like before and after the replace?

File: src/KpmRepoManager.ts [hunk 3]
```diff
@@ -1,185 +1,179 @@
...
-        // something went wrong, but don't try to parse a null or undefined str
-        return 0;
+  // get the list of users that modified this file
+  let resultList = await getCommandResult(cmd, projectDir);
+  if (!resultList) {
+    // something went wrong, but don't try to parse a null or undefined str
+    return 0;
+  }
+
+  if (resultList.length > 0) {
```
Comment (bjacobson26) [inline src/KpmRepoManager.ts:75]: how about: 

```
function countUnique(iterable) {
  return new Set(iterable).size;
}

countUnique(resultList)
```

**Comment by bjacobson26** [comment]
probably should also include updating the appjwt logic to match codetime: 
https://github.com/swdotcom/swdc-vscode-musictime/blob/360316bd82eed6889bc2ef99c121e60abd4eb3ad/src/music/MusicControlManager.ts#L826-L830

here is the code time commit: https://github.com/swdotcom/swdc-vscode/commit/061e514673eacd80bc1de186b898c1ea9a9ffb94

**Comment by xavluiz** [comment]
> probably should also include updating the appjwt logic to match codetime:
> https://github.com/swdotcom/swdc-vscode-musictime/blob/360316bd82eed6889bc2ef99c121e60abd4eb3ad/src/music/MusicControlManager.ts#L826-L830
> 
> here is the code time commit: [swdotcom/swdc-vscode@061e514](https://github.com/swdotcom/swdc-vscode/commit/061e514673eacd80bc1de186b898c1ea9a9ffb94)

updated

**Review by bjacobson26 — APPROVED** [review]

**Comment by xavluiz** [comment]
> should add /dist to the .gitignore

you're right. i noticed it getting checked in. will do

File: src/KpmRepoManager.ts (hunk 2 above)
Comment (xavluiz) [reply src/KpmRepoManager.ts:41]: will do

File: src/KpmRepoManager.ts (hunk 3 above)
Comment (xavluiz) [reply src/KpmRepoManager.ts:75]: i love this way of doing it. i'll replace that, thanks.

File: src/KpmRepoManager.ts (hunk 2 above)
Comment (xavluiz) [reply src/KpmRepoManager.ts]: moved this and other command related to the util and added comments.

File: src/KpmRepoManager.ts (hunk 3 above)
Comment (xavluiz) [reply src/KpmRepoManager.ts]: added that function, thanks!

**Pull request merged**